master/buildbot/www/config.py

```diff
@@ -39,11 +41,51 @@ def reconfigResource(self, new_config):
         if isinstance(self.config.get('versions'), list):
             versions += self.config['versions']
 
+        template_dir = self.config.get('custom_templates_dir')
```

Comment:
What about something more *dry & kiss* ?

```
self.custom_templates = {}
template_dir = self.config.pop('custom_templates_dir', None)
if template_dir:
    template_dir = os.path.join(self.master.basedir, template_dir)
    ...
```

*Avoiding `else` and `del` statements :)*